Scripts/should-run-smoke-test.sh

```
@@ -0,0 +1,57 @@
+
+# Folders where changes ALONE should NOT trigger the job
+ALLOWED_PREFIXES=(
+  "Azure_Pipelines/"
+  "docs/"
+  "Scripts/"
+  ".github/"
+  "Web/"
+)
+
+# Collect changed files (added/changed/renamed)
+mapfile -t FILES < <(git diff --name-only --diff-filter=ACMR ${DIFF_RANGE})
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)